### PR TITLE
Add From<String> trait on ModuleError

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: libchisel/src/lib.rs [hunk 1]
```diff
@@ -128,6 +134,12 @@ mod tests {
         assert!(result.is_ok());
     }
 
+    #[test]
+    fn from_error() {
+        let err: ModuleError = "custom message".to_string().into();
```
Comment (axic) [inline libchisel/src/lib.rs]: Not really useful in this form, but is when trying to unwrap from a function returning `Result<..., String>` into `Result<..., ModuleError>`

File: libchisel/src/lib.rs [hunk 2]
```diff
@@ -45,6 +45,12 @@ pub trait ModulePreset {
         Self: std::marker::Sized;
 }
 
+impl From<String> for ModuleError {
```
Comment (jakelang) [inline libchisel/src/lib.rs]: Could perhaps also implement `From<&'a str>` as well.

Comment (axic) [reply libchisel/src/lib.rs]: Do you want to implement that in a new PR? I'm for just merging this.

Comment (jakelang) [reply libchisel/src/lib.rs]: ok, sounds good

**Pull request merged**